### PR TITLE
[DOCS] Document ingest pipelines for Fleet and Elastic Agent

### PR DESCRIPTION
**Pull request by jrodewig**
Adds a section to the ingest tutorials covering Fleet and Elastic Agent.

### Preview
https://elasticsearch_70907.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/ingest.html#pipelines-for-fleet-elastic-agent

### CODE REVIEW

File: docs/reference/ingest.asciidoc [hunk 1]
```diff
@@ -266,6 +266,128 @@ Use the <<index-final-pipeline,`index.final_pipeline`>> index setting to set a
...
+`index.default_pipeline` index setting.
+
+[source,console]
+----
+# Creates a component template for index settings
+PUT _component_template/my-settings-component-template
+{
+  "template": {
+    "settings": {
+      "index.default_pipeline": "my-pipeline"
```
Comment (ruflin) [inline docs/reference/ingest.asciidoc:309]: I would have in our recommendation to name the pipeline the same as the data stream to just make it a habit.

**Review by ruflin — COMMENTED** [review]
Thank your for taking this on 🎉 @dedemorton We should make sure to link to this from the Fleet docs.

File: docs/reference/ingest.asciidoc (hunk 1 above)
Comment (jrodewig) [reply docs/reference/ingest.asciidoc]: Thanks @ruflin. I incorporated this naming convention into the docs with 5d584dc549799dc4a36be76debb578c30a6d2b4d and aeca6c3572fbc84ba8fa1eb2bd214ffcd62df0bf.

**Comment by jrodewig** [comment]
@dedemorton Let me know what you think of these changes. I'm also happy with housing this in Fleet Guide (feel free to take any of the copy here) and linking from here if that makes more sense.

Thanks again for your help and review!

**Comment by jrodewig** [comment]
I spoke to @dedemorton offline, and she's comfortable moving forward with these changes. We can always migrate this content over to the Fleet Guide later if wanted.

**Pull request merged**